selfservice/strategy/oidc/provider_linkedin.go

```diff
@@ -0,0 +1,150 @@
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/linkedin"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/ory/herodot"
+)
+
+type Profile struct {
+	LocalizedLastName string `json:"localizedLastName"`
+	ProfilePicture    struct {
+		DisplayImage string `json:"displayImage"`
+	} `json:"profilePicture"`
+	FirstName struct {
+		Localized struct {
+			EnUS string `json:"en_US"`
+		} `json:"localized"`
+		PreferredLocale struct {
+			Country  string `json:"country"`
+			Language string `json:"language"`
+		} `json:"preferredLocale"`
+	} `json:"firstName"`
+	LastName struct {
+		Localized struct {
+			EnUS string `json:"en_US"`
+		} `json:"localized"`
+		PreferredLocale struct {
+			Country  string `json:"country"`
+			Language string `json:"language"`
+		} `json:"preferredLocale"`
+	} `json:"lastName"`
+	ID                 string `json:"id"`
+	LocalizedFirstName string `json:"localizedFirstName"`
+}
+
+type EmailAddress struct {
+	Elements []struct {
+		Handle struct {
+			EmailAddress string `json:"emailAddress"`
+		} `json:"handle~"`
+		HandleUrn string `json:"handle"`
+	} `json:"elements"`
+}
+
+type APIUrl string
+
+const (
+	ProfileUrl APIUrl = "https://api.linkedin.com/v2/me"
+	EmailUrl          = "https://api.linkedin.com/v2/emailAddress?q=members&projection=(elements*(handle~))"
+)
+
+type ProviderLinkedIn struct {
+	config *Configuration
+	reg    dependencies
+}
+
+func NewProviderLinkedIn(
+	config *Configuration,
+	reg dependencies,
+) *ProviderLinkedIn {
+	return &ProviderLinkedIn{
+		config: config,
+		reg:    reg,
+	}
+}
+
+func (l *ProviderLinkedIn) Config() *Configuration {
+	return l.config
+}
+
+func (l *ProviderLinkedIn) oauth2(ctx context.Context) *oauth2.Config {
+	return &oauth2.Config{
+		ClientID:     l.config.ClientID,
+		ClientSecret: l.config.ClientSecret,
+		Endpoint:     linkedin.Endpoint,
+		Scopes:       l.config.Scope,
+		RedirectURL:  l.config.Redir(l.reg.Config(ctx).SelfPublicURL()),
+	}
+}
+
+func (l *ProviderLinkedIn) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	return l.oauth2(ctx), nil
+}
+
+func (l *ProviderLinkedIn) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
+	return []oauth2.AuthCodeOption{}
+}
+
+func (l *ProviderLinkedIn) ApiCall(url APIUrl, result interface{}, exchange *oauth2.Token) error {
+	var bearer = "Bearer " + exchange.AccessToken
+	req, err := http.NewRequest("GET", string(url), nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Authorization", bearer)
+	client := &http.Client{}
```

Comment:
If you merge with master you will find that the other clients use the registry to get a secure client. Please do that here too :)